sqlstore/src/main/java/uk/ac/bbsrc/tgac/miso/persistence/impl/HibernateSecurityDao.java

```diff
@@ -0,0 +1,225 @@
+@Repository
+public class HibernateSecurityDao implements SecurityStore {
+  private static final String USER_TABLE_NAME = "User";
+  private static final String GROUP_TABLE_NAME = "_Group";
+
+  protected static final Logger log = LoggerFactory.getLogger(HibernateSecurityDao.class);
+
+  @Autowired
+  private SessionFactory sessionFactory;
+
+  private JdbcTemplate template;
+
+  private Session currentSession() {
+    return getSessionFactory().getCurrentSession();
+  }
+
+  @Override
+  public Group getGroupById(Long groupId) throws IOException {
+    return (Group) currentSession().get(Group.class, groupId);
+  }
+  @Override
+  public Group getGroupByName(String groupName) throws IOException {
+    Criteria criteria = currentSession().createCriteria(Group.class);
+    criteria.add(Restrictions.eq("name", groupName));
+    return (Group) criteria.uniqueResult();
+  }
+
+  @Override
+  public Map<String, Integer> getGroupColumnSizes() throws IOException {
+    return DbUtils.getColumnSizes(template, GROUP_TABLE_NAME);
+  }
+
+  @CoverageIgnore
+  public JdbcTemplate getJdbcTemplate() {
+    return template;
+  }
+
+  @Override
+  public SecurityProfile getSecurityProfileById(Long profileId) throws IOException {
+    return (SecurityProfile) currentSession().get(SecurityProfile.class, profileId);
+  }
+
+  public SessionFactory getSessionFactory() {
+    return sessionFactory;
+  }
+
+  @Override
+  public User getUserByEmail(String email) throws IOException {
+    Criteria criteria = currentSession().createCriteria(UserImpl.class);
+    criteria.add(Restrictions.eq("email", email));
+    return (User) criteria.uniqueResult();
+  }
+
+  @Override
+  public User getUserById(Long userId) throws IOException {
+    return (User) currentSession().get(UserImpl.class, userId);
+  }
+
+  @Override
+  public User getUserByLoginName(String loginName) throws IOException {
+    Criteria criteria = currentSession().createCriteria(UserImpl.class);
+    criteria.add(Restrictions.eq("loginName", loginName));
+    return (User) criteria.uniqueResult();
+  }
+
+  @Override
+  public Map<String, Integer> getUserColumnSizes() throws IOException {
+    return DbUtils.getColumnSizes(template, USER_TABLE_NAME);
+  }
+
+  @Override
+  public Collection<Group> listAllGroups() throws IOException {
+    Criteria criteria = currentSession().createCriteria(Group.class);
+    @SuppressWarnings("unchecked")
+    List<Group> results = criteria.list();
+    return results;
+  }
+
+  @Override
+  public Collection<User> listAllUsers() throws IOException {
+    Criteria criteria = currentSession().createCriteria(UserImpl.class);
+    @SuppressWarnings("unchecked")
+    List<User> results = criteria.list();
+    return results;
+  }
+
+  @Override
+  public Collection<Group> listGroupsByIds(Collection<Long> groupIds) throws IOException {
+    List<Group> groups = new ArrayList<>(groupIds.size());
+    for (Long id : groupIds) {
+      groups.add(getGroupById(id));
```

Comment:
Should be able to do a multi-id query using something like
`criteria.add(Restrictions.in("groupId", groupIds))`

Current method allows for nulls to be included in the returned list too I think